platforms/chibios/GENERIC_STM32_F446XE/configs/mcuconf.h

```diff
@@ -0,0 +1,379 @@
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#ifndef MCUCONF_H
+#define MCUCONF_H
+
+/*
+ * STM32F4xx drivers configuration.
+ * The following settings override the default settings present in
+ * the various device driver implementation headers.
+ * Note that the settings for each driver only have effect if the whole
+ * driver is enabled in halconf.h.
+ *
+ * IRQ priorities:
+ * 15...0       Lowest...Highest.
+ *
+ * DMA priorities:
+ * 0...3        Lowest...Highest.
+ */
+
+#define STM32F4xx_MCUCONF
+
+/*
+ * HAL driver system settings.
+ */
+#define STM32_NO_INIT                       FALSE
+#define STM32_HSI_ENABLED                   FALSE
+#define STM32_LSI_ENABLED                   TRUE
+#define STM32_HSE_ENABLED                   TRUE
+#define STM32_LSE_ENABLED                   FALSE
+#define STM32_CLOCK48_REQUIRED              TRUE
+#define STM32_SW                            STM32_SW_PLL
+#define STM32_PLLSRC                        STM32_PLLSRC_HSE
+#if STM32_HSECLK == 8000000
+#    define STM32_PLLM_VALUE                    8
+#    define STM32_PLLN_VALUE                    360
+#    define STM32_PLLP_VALUE                    2
+#    define STM32_PLLQ_VALUE                    7
+#    define STM32_PLLI2SN_VALUE                 192
+#    define STM32_PLLI2SM_VALUE                 4
+#    define STM32_PLLI2SR_VALUE                 4
+#    define STM32_PLLI2SP_VALUE                 4
+#    define STM32_PLLI2SQ_VALUE                 4
+#    define STM32_PLLSAIN_VALUE                 192
+#    define STM32_PLLSAIM_VALUE                 4
+#    define STM32_PLLSAIP_VALUE                 8
+#    define STM32_PLLSAIQ_VALUE                 4
+#elif STM32_HSECLK == 16000000
+#    define STM32_PLLM_VALUE                    8
+#    define STM32_PLLN_VALUE                    180
+#    define STM32_PLLP_VALUE                    2
+#    define STM32_PLLQ_VALUE                    7
+#    define STM32_PLLI2SN_VALUE                 192
+#    define STM32_PLLI2SM_VALUE                 8
+#    define STM32_PLLI2SR_VALUE                 4
+#    define STM32_PLLI2SP_VALUE                 4
+#    define STM32_PLLI2SQ_VALUE                 4
+#    define STM32_PLLSAIN_VALUE                 192
+#    define STM32_PLLSAIM_VALUE                 8
+#    define STM32_PLLSAIP_VALUE                 8
+#    define STM32_PLLSAIQ_VALUE                 4
+#else
+#    error "Currently unsupported STM32_HSECLK"
+#endif
```

Comment:
This section will need to be "rolled back" to defaults as per ChibiOS default config template -- the normal ChibiOS upgrade script will obliterate these changes you've made (as per the template below), and QMK isn't in a position to reapply modifications for multiple MCUs after each ChibiOS upgrade.

See: <https://github.com/ChibiOS/ChibiOS/blob/master/tools/ftl/processors/conf/mcuconf_stm32f446xx/mcuconf.h.ftl>

Reply:
Keyboards that need to override this can do so with the normal "override format" with `#include_next <mcuconf.h>`.

Reply:
@tzarc Could you clarify which upgrade script you are referencing, so I can understand the flow?
I took the current mcuconf.h from lib/chibios/demos/STM32/RT-STM32F446RE-NUCLEO64/cfg/mcuconf.h , and I added my modifications on top. Should I just use the unmodified NUCLEO64/cfg/mcuconf.h ? or should I somehow generate a new mcuconf.h from the template file you mentioned?

Reply:
Also I wonder is there a good way I can share mcuconf.h overrides among multiple keyboards using the same controller?

Reply:
I have fixed the merge conflicts, I have reverted mcuconf.h to be the same as lib/chibios/demos/STM32/RT-STM32F446RE-NUCLEO64/cfg/mcuconf.h, and moved the customizations into my keyboard folder with an mcuconf override (which is coming as a PR soon too).
Let me know if this is okay.
Please note also that USB is disabled by default so all keyboards using this chip will have to do at least some form of mcuconf override, even if they use the same clock configuration as the nucleo board.

Reply:
I also want to point out that our "GENERIC_STM32_G474XE" also has customisations. As far as I can tell USB is disabled in chibios, but we enable it without using mcuconf overrides in our mcuconf.h copy. That's an issue, correct?

Reply:
>  Please note also that USB is disabled by default so all keyboards using this chip will have to do at least some form of mcuconf override, even if they use the same clock configuration as the nucleo board.

Please change this -- all of the board definitions in QMK are set up such that if someone were to try to design a keyboard (with no special features other than keypresses), then they would not need to provide any configurations at all.

Reply:
For clarity, the choice to maintain a small set of config files was a conscious decision -- we have upwards of 150 ChibiOS boards, and this "cut-down" config format allows us to decouple keyboards from ChibiOS _versions_, due to ChibiOS configs being versioned.

With the configs living under `platforms/chibios/`, we're able to maintain and update 10-15 files when upgrading ChibiOS, instead of 600+ as was the case previously.

Reply:
I'm confused, cause I'm getting conflicting information here. On one hand you say I can't have customisations in the `platforms/chibios/*/configs/mcuconf.h` file compared to what's in chibios, on the other hand you are saying that usb must be enabled. Could you clarify?

Reply:
Discussed on Discord, all good now.

Reply:
Alright, so for third-party readers benefit, we moved the conversation to discord, where we clarified, that it's okay to set defaults in `platforms/chibios/*/configs/mcuconf.h` that are different from chibios defaults. What is not okay is to change the file format by adding #ifdefs and such, and this is because the mcuconf.h file is parsed by a chibios upgrade script, and cannot parse #ifdef statements.

Also it is a requirement that the qmk version of mcuconf.h would enable USB and disable other peripherals.

Which clocking setup is supported by default is my choice since I'm the first user of this mcu in qmk. I will configure 16MHz crystal support, since this is what my controller board uses, that will apply to at least 6 keyboards, so having my defaults in mcuconf.h will eliminate code duplication, and also this is the only clock configuration I can acutally test.